### ISSUE TITLE
[java] hashCode() with return type != int causes native compilation error

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#9059]
This code is valid from a Haxe perspective:

```haxe
class Main {
	static function main() {}

	public function hashCode():String {
		return "";
	}
}
```

But fails to compile to Java - I guess the field name needs to be escaped if you have a `hashCode` that does not return an integer.

```
javac.exe "-sourcepath" "src" "-d" "obj" "-g" "@cmd"
source/Main.hx:4: lines 4-6 : A hashCode() function should return an Int!
src\haxe\root\Main.java:40: error: hashCode() in Main cannot override hashCode() in Object
        public java.lang.String hashCode()
                                ^
  return type String is not compatible with int
1 error
Compilation error
Native compilation failed
```

The JVM target doesn't seem to care though.

**Comment by Aurel300** [comment]
Should probably be added to reserved keywords?

**Comment by Gama11** [comment]
Well, people might rely on it being generated as-is to interop with native Java code that expects a `hashCode():Int` implementation.

**Comment by ibilon** [comment]
The chances are pretty low but wouldn't that pose problems if you had a `hashCode():Int` that didn't return an actual hashcode? or if you didn't want java to use it maybe, not sure.
I mean yeah like really really slim chances, but maybe a java specific metadata would be better if you wanted your function to be used as native hashCode?

**Comment by Gama11** [comment]
Hm, I guess `@:native("hashCode")` would probably already work. But since that's currently not needed it seems, it'd be a breaking change.

**Comment by Simn** [comment]
The Java target will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
